### DISCUSSION TITLE
Change column type in T4 model generation

### DISCUSSION THREAD

**Discussion by CatalinCernea** [linq2db/linq2db#3638] [General]
I was wondering if it was possible to change the column type from decimal to float in the T4 template model generation in order to have the correspoding class with double and not decimal on specific columns.
I tryied adding the following after LoadMetaData and before GenerateModel but still get generated as decimal.
Is it even possible?

```
foreach (var t in Tables.ToList())
{
  foreach(var c in t.Value.Columns.ToList())
  {
	  if (c.Value.ColumnType.Contains("decimal"))
	  {
		  c.Value.ColumnType = "float"; 
	  }
  }
}
```

**Comment by MaceWindu** [comment]
Check `Change column property type` example here https://linq2db.github.io/articles/T4.html#example-of-generation-process-customization

**Reply by CatalinCernea** [reply]
I saw those examples thats why I posted the code above. 
I took example from there and the code runs but still I get the column generated as decimal.
What I'm doing wrong?

**Reply by MaceWindu** [reply]
There is important note: `Use the following code to modify your model before you call the GenerateModel() method.`
If you modify model after GenerateModel call, changes will be ignored
